File: backend/app/Models/Deck.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Concerns\HasUuids;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Deck extends Model
{
    use HasFactory, HasUuids;

    protected $fillable = [
        'name',
        'language',
        'description',
        'is_starter',
    ];
    
    public function cards()
    {
        return $this->hasMany(Card::class);
    }
}

File: shared/src/commonMain/kotlin/com/vocapop/domain/models/Deck.kt
package com.vocapop.domain.models

import kotlinx.serialization.Serializable

@Serializable
data class Deck(
    val id: String,
    val name: String,
    val language: String, // e.g., "es", "fr"
    val description: String? = null,
    val isStarter: Boolean = false,
    val cardCount: Int = 0,
    val learnedCount: Int = 0
)
